universe/Conditions.cpp

```diff
@@ -9385,6 +9675,24 @@ unsigned int ValueTest::GetCheckSum() const {
     return retval;
 }
 
+std::unique_ptr<Condition> ValueTest::Clone() const {
+    auto retval = std::make_unique<ValueTest>(ValueRef::CloneUnique(m_value_ref1),
```

Comment:
for cases like this, I'd rather add a new private constructor that takes no parameters and doesn't init the invariants. it seems odd and assymmetric to use the 3-int constructor, which does all the invariant initialization checks for those input valuerefs, and then overwrite that by copying from `this` anyway.

Reply:
Not sure why, but `std::make_unique` somehow doesn't accept private constructors.

Reply:
I was hoping one could make `std::make_unique` a friend function of `ValueTest` but that apparently isn't portably possible due to STD implementation details.

Reply:
Made it

Reply:
from edit:

> I'd also add a protected copy constructor to base class `Condition`, which when defaulted should just copy the invariant members: